system-addon/lib/PingCentre.jsm

```diff
@@ -0,0 +1,149 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const {interfaces: Ci, utils: Cu} = Components;
+Cu.import("resource://gre/modules/Preferences.jsm");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+Cu.importGlobalProperties(["fetch"]);
+
+XPCOMUtils.defineLazyModuleGetter(this, "ClientID",
+  "resource://gre/modules/ClientID.jsm");
+XPCOMUtils.defineLazyModuleGetter(this, "console",
+  "resource://gre/modules/Console.jsm");
+
+const PREF_BRANCH = "browser.ping-centre.telemetry.";
+
+const TELEMETRY_PREF = `${PREF_BRANCH}telemetry`;
+const LOGGING_PREF = `${PREF_BRANCH}log`;
+const IS_PRODUCTION_PREF = `${PREF_BRANCH}production`;
```

Comment:
Instead of using this strategy, maybe something like this would be easier?
- define some endpoint(s) for different builds (nightly, release, beta). You can check what kind of built it is with [this file](http://searchfox.org/mozilla-central/source/toolkit/modules/AppConstants.jsm)
- take an optional override pref name